### PR TITLE
LF: drop support from deprecated LF value versions. (#8083)

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This advances the state of #7788.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/spec/value.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@
 DAML-LF Value Specification
 ===========================
 
-**version 5, 29 May 2019**
+**version 6, 10 Dev 2020**
```
Comment (hurryabit) [inline daml-lf/spec/value.rst:7]:
```suggestion
**version 6, 10 Dec 2020**
```

File: daml-lf/spec/value.rst [hunk 2]
```diff
@@ -146,18 +146,18 @@ Version history
 This table lists every version of this specification in ascending order
 (oldest first).
 
+Support for value versions 1 to 4 was dropped on 2020-12-10.
```
Comment (hurryabit) [inline daml-lf/spec/value.rst:149]: Didn't we drop version 5 today as well?

File: ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  private[translation] def handleDeprecatedValueVersions[X](
+      x: Either[ValueCoder.DecodeError, X],
+  ): Either[ValueCoder.DecodeError, X] = {
+    x match {
+      case DeprecatedValueVersionsError(deprecateVersion) =>
+        logger.error(
+          s"*** Deserialization of value version $deprecateVersion is not supported by the SDK 1.9.0 or latter. ***")
+        logger.error(
+          s"*** Please upgrade your sandbox database by upgrading your SDK to 1.7 or 1.8 first. ***")
+    }
```
Comment (remyhaemmerle-da) [inline ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala:40]:
```suggestion
    }
```

File: ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/ValueSerializerTest.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      val negativeTestCases = Table("version", "0", "6", "7", "11", "999", "some version")
+      val positiveTestCases = Table("version", "1", "2", "3", "4", "5")
+      val valueBuilder =
+        proto.VersionedValue.newBuilder().setValue(proto.Value.newBuilder().setBool(true))
+
+      forEvery(negativeTestCases) { version =>
+        val x = ValueSerializer.DeprecatedValueVersionsError.unapply(
+          ValueCoder.decodeValue(ValueCoder.CidDecoder, valueBuilder.setVersion(version).build())
+        )
+        remy.log(version -> x)
```
Comment (gerolf-da) [inline ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/ValueSerializerTest.scala:27]: `remy.log` :trollface:

File: ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala (hunk 3 above)
Comment (remyhaemmerle-da) [inline ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala:40]: @gerolf-da Would it be better to throw a error or add a `System.exit(1)` here

File: ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  private[translation] def handleDeprecatedValueVersions[X](
+      x: Either[ValueCoder.DecodeError, X],
+  ): Either[ValueCoder.DecodeError, X] = {
+    x match {
+      case DeprecatedValueVersionsError(deprecateVersion) =>
+        logger.error(
+          s"*** Deserialization of value version $deprecateVersion is not supported by the SDK 1.9.0 or latter. ***")
+        logger.error(
+          s"*** Please upgrade your sandbox database by upgrading your SDK to 1.7 or 1.8 first. ***")
```
Comment (remyhaemmerle-da) [inline ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala:39]: @gerolf-da Would it be better to throw a error or add a System.exit(1) here

File: ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      }
+  }
+
+  private[translation] def handleDeprecatedValueVersions[X](
+      x: Either[ValueCoder.DecodeError, X],
+  ): Either[ValueCoder.DecodeError, X] = {
+    x match {
+      case DeprecatedValueVersionsError(deprecateVersion) =>
+        logger.error(
+          s"*** Deserialization of value version $deprecateVersion is not supported by the SDK 1.9.0 or latter. ***")
```
Comment (gerolf-da) [inline ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala:37]:
```suggestion
      case DeprecatedValueVersionsError(deprecatedVersion) =>
        logger.error(
          s"*** Deserialization of value version $deprecatedVersion is not supported by the SDK 1.9.0 or later. ***")
```

**Review by hurryabit — COMMENTED** [review]
The changes in the `daml-lf` directory look good to me. I'll leave the approval to whoever reviews the changes in `ledger`.

File: ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/ValueSerializerTest.scala [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        x shouldBe None
+      }
+
+      forEvery(positiveTestCases) { version =>
+        val x = ValueSerializer.DeprecatedValueVersionsError.unapply(
+          ValueCoder.decodeValue(ValueCoder.CidDecoder, valueBuilder.setVersion(version).build())
+        )
+
+        remy.log(version -> x)
```
Comment (gerolf-da) [inline ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/ValueSerializerTest.scala:36]: `remy.log` :trollface:

**Review by gerolf-da — APPROVED** [review]
Looks good to me as well. Thanks @remyhaemmerle-da!

File: ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala (hunk 5 above)
Comment (gerolf-da) [reply ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala:39]: Never ever use `System.exit` in a library. You don't know where the library is used. I'd rather not kill an unknown process.

File: daml-lf/spec/value.rst (hunk 2 above)
Comment (remyhaemmerle-da) [inline daml-lf/spec/value.rst:149]:
```suggestion
Support for value versions 1 to 5 was dropped on 2020-12-10.
```

File: ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala (hunk 5 above)
Comment (remyhaemmerle-da) [reply ledger/participant-integration-api/src/main/scala/db/migration/translation/ValueSerializer.scala]: :+1:

File: daml-lf/spec/value.rst (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/value.rst]: fixed.

**Pull request merged**